be/src/test/java/com/jjikmuk/sikdorak/acceptance/DocumentFormatGenerator.java

```diff
@@ -0,0 +1,35 @@
+package com.jjikmuk.sikdorak.acceptance;
+
+import static org.springframework.restdocs.payload.PayloadDocumentation.fieldWithPath;
+import static org.springframework.restdocs.snippet.Attributes.key;
+
+import org.springframework.restdocs.constraints.ConstraintDescriptions;
+import org.springframework.restdocs.payload.FieldDescriptor;
+import org.springframework.restdocs.payload.JsonFieldType;
+import org.springframework.restdocs.snippet.Attributes.Attribute;
+
+public interface DocumentFormatGenerator {
+
+	FieldDescriptor DEFAULT_CODE_RESPONSE_FIELD_DESCRIPTOR =
+		fieldWithPath("code")
+			.type(JsonFieldType.STRING)
+			.description("식도락 서버 전용 상태코드");
+
+	FieldDescriptor DEFAULT_MESSAGE_RESPONSE_FIELD_DESCRIPTORS =
+		fieldWithPath("message")
+			.type(JsonFieldType.STRING)
+			.description("자세한 응답 메세지");
+
+	static FieldDescriptor fieldWithPathAndValidConstraints(String path, Class<?> clazz) {
+		FieldDescriptor fieldWithPath = fieldWithPath(path);
+
+		ConstraintDescriptions constraintDescriptions = new ConstraintDescriptions(clazz);
+		Attribute constraints = key("constraints").value(
+			constraintDescriptions.descriptionsForProperty(path));
+
+		fieldWithPath.attributes(constraints);
+
+		return fieldWithPath;
+	}
```

Comment:
p2;
알고보니 가면인자를 뒤에두고, 앞에 다른 매개변수를 두면 같이 쓰는게 가능하더라구요.
ReviewSnippet 에서 코드 중복이 있는 것 같은데, 아래와 같은 형태로 코드 중복을 줄일 수 있는 형태로 가는건 어떨까요?

```suggestion
static Snippet commonResponseFieldsWithValidConstraints(Class<?> clazz, FieldDescriptor ...fields) {
		final String dataPrefix = "data.[].";

		List<FieldDescriptor> commonResponseFields = List.of(
				DEFAULT_CODE_RESPONSE_FIELD_DESCRIPTOR,
				DEFAULT_MESSAGE_RESPONSE_FIELD_DESCRIPTORS
		);

		for (FieldDescriptor field : fields) {
			commonResponseFields.add(
					fieldWithPath(dataPrefix + field.getPath())
							.type(field.getType())
							.description(field.getDescription())
							.attributes(getFieldConstraints(clazz, field.getPath()))
			);
		}

		return responseFields(commonResponseFields);
	}

	static Attribute getFieldConstraints(Class<?> clazz, String path) {
		FieldDescriptor fieldWithPath = fieldWithPath(path);

		ConstraintDescriptions constraintDescriptions = new ConstraintDescriptions(clazz);
		return key("constraints").value(
				constraintDescriptions.descriptionsForProperty(path));
	}
```

Reply:
위와 같은 형식으로 한다면 `ReviewSnippet` 에서 `commonResponseFieldsWithValidConstraints` 를 사용하여 아래와 같이 코드를 작성할 수 있습니다.

```java
Snippet REVIEW_INSERT_REQUEST_SNIPPET = commonResponseFieldsWithValidConstraints(ReviewInsertRequest.class,
            fieldWithPath("reviewContent").type(JsonFieldType.STRING).description("리뷰 내용"),
            fieldWithPath("storeId").type(JsonFieldType.NUMBER).description("가게 아이디"),
            fieldWithPath("reviewScore").type(JsonFieldType.NUMBER).description("리뷰 점수"),
            fieldWithPath("reviewVisibility").type(JsonFieldType.STRING).description("리뷰 게시물의 공개 범위"),
            fieldWithPath("visitedDate").type(JsonFieldType.STRING).description("가게 방문일"),
            fieldWithPath("tags").type(JsonFieldType.ARRAY).description("리뷰를 표현하는 태그들"),
            fieldWithPath("images").type(JsonFieldType.ARRAY).description("리뷰를 위한 사진 URL")
    );
```

Reply:
반영 완료!!!